packages/xpetra/src/Map/Xpetra_MapFactory_decl.hpp

```diff
@@ -0,0 +1,586 @@
+//                  Copyright 2012 Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact
+//                    Jonathan Hu       (jhu@sandia.gov)
+//                    Andrey Prokopenko (aprokop@sandia.gov)
+//                    Ray Tuminaro      (rstumin@sandia.gov)
+//
+// ***********************************************************************
+//
+// @HEADER
+#ifndef XPETRA_MAPFACTORY_DECL_HPP
+#define XPETRA_MAPFACTORY_DECL_HPP
+
+#include "Xpetra_ConfigDefs.hpp"
+#include "Xpetra_Map.hpp"
+
+#ifdef HAVE_XPETRA_TPETRA
+#    include "Xpetra_TpetraMap.hpp"
+#endif
+#ifdef HAVE_XPETRA_EPETRA
+#    include "Xpetra_EpetraMap.hpp"
+#endif
+
+#include "Xpetra_Exceptions.hpp"
+
+namespace Xpetra {
+
+/// \class MapFactory
+/// \brief Create an Xpetra::Map instance.
+///
+/// Users must specify the exact class of the object that they want
+/// to create (either an Xpetra::TpetraMap or an Xpetra::EpetraMap).
+template<class LocalOrdinal,
+         class GlobalOrdinal,
+         class Node = typename Map<LocalOrdinal, GlobalOrdinal>::node_type>
+class MapFactory
+{
+
+
+  private:
+
+
+    //! Private constructor. This is a static class.
+    MapFactory() {}
+
+
+  public:
+
+
+    //! Map constructor with Xpetra-defined contiguous uniform distribution.
+#ifdef TPETRA_ENABLE_DEPRECATED_CODE
```

Comment:
This is an Xpetra thing, but you're using a Tpetra macro here.  Is that OK?  What if Tpetra is not enabled?

Reply:
@mhoemmen the `TPETRA_ENABLE_DEPRECATED_CODE` stuff isn't mine... that was already in develop when I forked my branch @tjfulle ?